include/nuttx/arch.h

```diff
@@ -308,9 +308,26 @@ int up_use_stack(FAR struct tcb_s *tcb, FAR void *stack, size_t stack_size);
  *
  *   - adj_stack_size: Stack size after removal of the stack frame from
  *     the stack
- *   - adj_stack_ptr: Adjusted initial stack pointer after the frame has
- *     been removed from the stack.  This will still be the initial value
- *     of the stack pointer when the task is started.
+ *   - adj_stack_ptr: Adjusted stack base pointer after the TLS Data and
+ *     Arguments has been removed from the stack allocation.
+ *
+ *   Here is the diagram after some allocation(tls, arg):
+ *
+ *                   +-------------+ <-stack_alloc_ptr(lowest)
+ *                   |  TLS Data   |
+ *                   +-------------+
+ *                   |  Arguments  |
+ *   adj_stack_ptr-> +-------------+\
+ *                   |  Available  | +
+ *                   |    Stack    | |
+ *                |  |             | |
+ *                |  |             | |
+ *                v  |             | +->adj_stack_size
+ *                   |             | |
+ *                   |             | |
+ *                   +-------------+ |
+ *                   | Thread Data | +
+ *                   +-------------+/
```

Comment:
Could you add this to the docs as well (for the `up_use_stack` function)? I think it is critical to understand all the variables.

Reply:
Ok, add to here:
https://github.com/apache/incubator-nuttx/pull/3517/files#diff-b0e4e9a1378b2a5ded4af14faa20cd4fea7cc12ae8d832df45817891718734eaR126-R142

Reply:
> 
> 
> Ok, add to here:
> https://github.com/apache/incubator-nuttx/pull/3517/files#diff-b0e4e9a1378b2a5ded4af14faa20cd4fea7cc12ae8d832df45817891718734eaR126-R142

Is that correct?  Isn't "Thread Data" at the other end?  Like the version I added here: https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=152115079

Reply:
Yes, "Thread Data" make confusion here. I will remove it, thanks.

Reply:
Done. I also copy the nice diagram of tls and task from wiki:
https://github.com/apache/incubator-nuttx/pull/3517/files#diff-2c833ffaa18374fab0d57499dc593937167b198a630cb4e9068f654aaebdc42dR74-R93